### PR TITLE
Group 6 - Refactoring Library class

### PR DESCRIPTION
**Pull request by PtrBld**

### CODE REVIEW

File: library.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
+from typing import Optional
+from library.model.book import Book
+from library.model.genre import Genre
+
+
+class Library():
+
+    # Prints crime audio books
```
Comment (knoerrj) [inline library.py:8]: Add input and output of method in header

File: library.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        print("*********************************************")
+        print("********* Longest Crime Audio Book **********")
+        print("*********************************************")
+
+        for b in self.get_books():
+            if Genre.CRIME in b.genres:
+                if b._book_type == "Audio":
+                    if max is not None and max.duration < b.duration:
+                        max = b
+                    # end if
```
Comment (knoerrj) [inline library.py:23]: Remove #end if or #end else

File: library.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+                # end if
+            else:
```
Comment (knoerrj) [inline library.py:25]: Remove whole else clause

File: library.py [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+        # print banner
+        print("*********************************************")
+        print("********* Longest Crime Audio Book **********")
+        print("*********************************************")
+
+        for b in self.get_books():
+            if Genre.CRIME in b.genres:
+                if b._book_type == "Audio":
+                    if max is not None and max.duration < b.duration:
```
Comment (knoerrj) [inline library.py:21]: Potentially stack if-clauses together into one

File: library.py [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+            else:
+                pass
+            # end else
+        # end for
+
+        print(f"Title: {max.title}")
+        print(f"Author: {max.authors}")
+        print(f"Duration: {max.duration}")
+        print(f"______________________________")
+        return str(max)
```
Comment (knoerrj) [inline library.py:34]: Is return str(max) needed?

File: library.py [hunk 6]
```diff
@@ -0,0 +1,43 @@
+from typing import Optional
+from library.model.book import Book
+from library.model.genre import Genre
+
+
+class Library():
```
Comment (knoerrj) [inline library.py:6]: Add class description

File: library.py [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+        print(f"Author: {max.authors}")
+        print(f"Duration: {max.duration}")
+        print(f"______________________________")
+        return str(max)
+
+    # Constructor
+    def __init__(self, books: list[Book]):
+        self._books = books
+
+    _books: list[Book] = []
```
Comment (knoerrj) [inline library.py:40]: What the purpose of this line?

File: library.py [hunk 8]
```diff
@@ -0,0 +1,43 @@
+from typing import Optional
+from library.model.book import Book
+from library.model.genre import Genre
+
+
+class Library():
+
+    # Prints crime audio books
+    def print_longest_crime_audio_book(self) -> str:
+        crime_audio_books: list[Book] = []
```
Comment (knoerrj) [inline library.py:10]: This variable is never used again

File: library.py (hunk 4 above)
Comment (knoerrj) [inline library.py:21]: if max is not none won't work. Set an initial value for max

File: library.py (hunk 1 above)
Comment (knoerrj) [reply library.py:8]: Plus the comment doesn't match the method functionality

File: library.py [hunk 9]
```diff
@@ -0,0 +1,43 @@
+from typing import Optional
+from library.model.book import Book
+from library.model.genre import Genre
+
+
+class Library():
+
+    # Prints crime audio books
+    def print_longest_crime_audio_book(self) -> str:
```
Comment (DeSl11) [inline library.py:9]: Name could be shorter

File: library.py [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+        crime_audio_books: list[Book] = []
+        max: Optional[Book] = None
+
+        # print banner
+        print("*********************************************")
+        print("********* Longest Crime Audio Book **********")
+        print("*********************************************")
+
+        for b in self.get_books():
+            if Genre.CRIME in b.genres:
```
Comment (knoerrj) [inline library.py:19]: Perhaps validate the ordering of the if-clauses for computational efficiency

**Pull request closed without merging**